de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -229,6 +245,99 @@ protected void dragTreeItemToDiagramEditor(SWTBotTreeItem item, SWTBotGefEditor
 	}
 	
 	/**
+	 * Returns the absolute rectangular bounds for a SWTBotGefEditpart element
+	 * @param SWTBotEditPart edit part to get bounds for
+	 * @return Rectangle bounds for SWTBotGefEditPart
+	 */
+	protected Rectangle getBoundsForEditPart(SWTBotGefEditPart swtBotEditPart) {
+		EditPart part = null;		
+		for (Field f : swtBotEditPart.getClass().getDeclaredFields()) {
+			if ("part".equals(f.getName())) {
+				// Here we're bypassing Java's OO-Security model, which is generally not advisable. It's meant to be a workaround to access
+				// otherwise inaccessible fields.
+				f.setAccessible(true);
+				try {
+					part = (EditPart) f.get(swtBotEditPart);
+				} catch (IllegalArgumentException | IllegalAccessException e) {
+					Activator.getDefault().getLog().log(new Status(Status.ERROR, Activator.getPluginId(), Status.ERROR, "Can not access SWTBotEditPart element or do a proper cast to EditPart type", e));
+				}
+			}
+		}		
+		IFigure figure = ((GraphicalEditPart) part).getFigure();
+		Rectangle bounds = figure.getBounds().getCopy();
+		figure.translateToAbsolute(bounds);		
+		return bounds;
+	}
+	
+	/**
+	 * We can only drop tree items to canvas elements. SWTBotGefViewer's canvas element is not 
+	 * capable of beeing accessed directly from external classes though. Further more SWTBotGefViewer's 
+	 * canvas member gets populated during runtime only. Hence we use reflections to get a hold on it here.
+	 * 
+	 * @param item Tree item that is beeing dragged
+	 * @param diagramEditor Graphiti diagram editor which the tree item is beeing dragged onto
+	 */
+	protected void dragTreeItemToDiagramEditor(SWTBotTreeItem item, SWTBotGefEditor diagramEditor, int x, int y) {
+		SWTBotGefViewer viewer = diagramEditor.getSWTBotGefViewer();
+		SWTBotGefFigureCanvas canvas = null;
+		
+		for (Field f : viewer.getClass().getDeclaredFields()) {
+			if (SWTBOT_CANVAS_FIELD_REFLECTION_NAME.equals(f.getName())) {

```

Comment:
There seems to be duplicate logic in various variants of this method.
Maybe we can unify this a bit.